src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -115,132 +95,19 @@ master_modify_multiple_shards(PG_FUNCTION_ARGS)
 		relationId = RangeVarGetRelid(updateStatement->relation, NoLock, failOK);
 		EnsureTablePermissions(relationId, ACL_UPDATE);
 	}
-	else if (IsA(queryTreeNode, TruncateStmt))
-	{
-		TruncateStmt *truncateStatement = (TruncateStmt *) queryTreeNode;
-		List *relationList = truncateStatement->relations;
-		RangeVar *rangeVar = NULL;
-
-		if (list_length(relationList) != 1)
-		{
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("master_modify_multiple_shards() can truncate only "
-							"one table")));
-		}
-
-		rangeVar = (RangeVar *) linitial(relationList);
-		relationId = RangeVarGetRelid(rangeVar, NoLock, failOK);
-		if (rangeVar->schemaname == NULL)
-		{
-			Oid schemaOid = get_rel_namespace(relationId);
-			char *schemaName = get_namespace_name(schemaOid);
-			rangeVar->schemaname = schemaName;
-		}
-
-		EnsureTablePermissions(relationId, ACL_TRUNCATE);
-	}
 	else
 	{
-		ereport(ERROR, (errmsg("query \"%s\" is not a delete, update, or truncate "
+		ereport(ERROR, (errmsg("query \"%s\" is not a delete or update "
 							   "statement", ApplyLogRedaction(queryString))));
 	}
 
 	CheckDistributedTable(relationId);
 
-	queryTreeList = pg_analyze_and_rewrite(rawStmt, queryString, NULL, 0, NULL);
-	modifyQuery = (Query *) linitial(queryTreeList);
-
-	operation = modifyQuery->commandType;
-	if (operation != CMD_UTILITY)
-	{
-		bool multiShardQuery = true;
-		DeferredErrorMessage *error =
-			ModifyQuerySupported(modifyQuery, modifyQuery, multiShardQuery, NULL);
-
-		if (error)
-		{
-			RaiseDeferredError(error, ERROR);
-		}
-
-		taskType = MODIFY_TASK;
-	}
-	else
-	{
-		taskType = DDL_TASK;
-	}
-
-	/* reject queries with a returning list */
-	if (list_length(modifyQuery->returningList) > 0)
-	{
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("master_modify_multiple_shards() does not support RETURNING")));
-	}
-
-	ExecuteMasterEvaluableFunctions(modifyQuery, NULL);
-
-	restrictClauseList = WhereClauseList(modifyQuery->jointree);
+	ereport(WARNING, (errmsg("master_modify_multiple_shards is deprecated and will be "
+							 "removed in a future release."),
+					  errhint("Run the command directly")));
 
-	prunedShardIntervalList =
-		PruneShards(relationId, tableId, restrictClauseList, NULL);
-
-	CHECK_FOR_INTERRUPTS();
-
-	taskList =
-		ModifyMultipleShardsTaskList(modifyQuery, prunedShardIntervalList, taskType);
-
-	if (MultiShardConnectionType == SEQUENTIAL_CONNECTION)
-	{
-		affectedTupleCount =
-			ExecuteModifyTasksSequentiallyWithoutResults(taskList, operation);
-	}
-	else
-	{
-		affectedTupleCount = ExecuteModifyTasksWithoutResults(taskList);
-	}
+	ExecuteQueryStringIntoDestReceiver(queryString, NULL, None_Receiver);
```

Comment:
Just to clarify for myself, we cannot have parameters passed to `master_modify_multiple_shards` so we can safely pass NULL?

Reply:
correct